pkg/pool/pool.go

```diff
@@ -61,10 +66,21 @@ func NewWithCapacity(workers, capacity int) Interface {
 		go func() {
 			for work := range i.workCh {
 				func() {
+					done := make(chan bool, 1)
 					defer i.wg.Done()
-					if err := work(); err != nil {
+					go func() {
```

Comment:
Hmm, we won't actually abort the work here, will we? It'll continue to run in the background.

Should we pass a `Context` to `work` and implement the timeout through that context? That gives the `work` pieces the opportunity to exit early, once the Context exists and vastly simplifies code here. After work returns, we'd check if the context error is still nil and only in that case return the actual result.

WDYT?

Reply:
I tried to change, but there are two difficulties.

- `work()` is an argument of `impl.Go()` and it is an implementation of errgroup, so `Context` cannot be passed to work() directory.
- When `work()`() gets timeout, it cannot return, so the timeout needs to be implemented in out side of the work().

I updated and implemented the timeout in `wg.Go`. But this might be different what you thought.

Reply:
I cannot find the benefits. But, I found a simple change for this and updated the PR.